tests/integration-service/status-reporting-to-pullrequest.go

```diff
@@ -200,6 +252,99 @@
+			sampleImage := "quay.io/redhat-appstudio/sample-image@sha256:841328df1b9f8c4087adbdcfec6cc99ac8308805dea83f6d415d6fb8d40227c1"
+			snapshotPush, err = f.AsKubeAdmin.IntegrationController.CreateSnapshotWithImage(componentName, applicationName, testNamespace, sampleImage)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		When("An snapshot of push event is created", func() {
+			It("checks if the global candidate is updated after push event", func() {
+				timeout = time.Second * 600
+				interval = time.Second * 10
+				Eventually(func() error {
+					snapshotPush, err = f.AsKubeAdmin.IntegrationController.GetSnapshot(snapshotPush.Name, "", "", testNamespace)
+					Expect(err).ShouldNot(HaveOccurred())
+
+					latestComponent, err := f.AsKubeAdmin.HasController.GetComponentByApplicationName(applicationName, testNamespace)
+					Expect(err).ShouldNot(HaveOccurred())
+					Expect(latestComponent.Spec.ContainerImage).ToNot(Equal(component.Spec.ContainerImage))
+					return nil
+
+				}, timeout, interval).Should(Succeed(), fmt.Sprintf("time out when waiting for updating the global candidate in %s namespace", testNamespace))
+			})
+
+			It("checks if all of the integrationPipelineRuns created by push event passed", Label("slow"), func() {
+				Expect(f.AsKubeAdmin.IntegrationController.WaitForAllIntegrationPipelinesToBeFinished(testNamespace, applicationName, snapshotPush)).To(Succeed(), "Error when waiting for one of the integration pipelines to finish in %s namespace", testNamespace)
+			})
+
+			It("checks if a Release is created successfully", func() {
+				timeout = time.Second * 60
+				interval = time.Second * 5
+				Eventually(func() error {
+					_, err := f.AsKubeAdmin.ReleaseController.GetReleases(testNamespace)
+					return err
+				}, timeout, interval).Should(Succeed(), fmt.Sprintf("time out when waiting for release created for snapshot %s/%s", snapshotPush.GetNamespace(), snapshotPush.GetName()))
+			})
 		})
 	})
 })
+
+func createApp(f framework.Framework, testNamespace string) string {
+	applicationName := fmt.Sprintf("integ-app-%s", util.GenerateRandomString(4))
+
+	_, err := f.AsKubeAdmin.HasController.CreateApplication(applicationName, testNamespace)
+	Expect(err).NotTo(HaveOccurred())
+
+	return applicationName
+}
+
+func cleanup(f framework.Framework, testNamespace, applicationName, componentName string) {
+	if !CurrentSpecReport().Failed() {
+		Expect(f.AsKubeAdmin.HasController.DeleteApplication(applicationName, testNamespace, false)).To(Succeed())
+		Expect(f.AsKubeAdmin.HasController.DeleteComponent(componentName, testNamespace, false)).To(Succeed())
+		integrationTestScenarios, err := f.AsKubeAdmin.IntegrationController.GetIntegrationTestScenarios(applicationName, testNamespace)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		for _, testScenario := range *integrationTestScenarios {
+			Expect(f.AsKubeAdmin.IntegrationController.DeleteIntegrationTestScenario(&testScenario, testNamespace)).To(Succeed())
```

Comment:
should this be addressed?

Reply:
it was [there in the code already](https://github.com/konflux-ci/e2e-tests/blob/995cc017c91794609896078f18141e9596e48678/tests/integration-service/integration.go#L376), haven't changed it, I just copied here.

Reply:
Reverted back the change, could not remember earlier the why it is used, it is needed to used in [this test](https://github.com/konflux-ci/e2e-tests/pull/1309/files#diff-938ddf0d569d68c3df496d8b4065614ff56cc303c33555755621cffcec97ef74R1170) to avoid conflict with sample repo used in [this test](https://github.com/konflux-ci/e2e-tests/pull/1309/files#diff-938ddf0d569d68c3df496d8b4065614ff56cc303c33555755621cffcec97ef74R43)